sys/include/net/nanocoap.h

```diff
@@ -1039,17 +1065,18 @@ static inline size_t coap_opt_put_block2(uint8_t *buf, uint16_t lastonum,
 }
 
 /**
- * @brief   Insert block option into buffer from block struct
+ * @brief   Encode the given uint option into buffer
  *
- * @param[in]   buf         buffer to write to
- * @param[in]   lastonum    last option number (must be < @p option)
- * @param[in]   block       block option attribute struct
- * @param[in]   option      option number (block1 or block2)
+ * @param[out]  buf         buffer to write to
+ * @param[in]   lastonum    number of previous option (for delta calculation),
+ *                          or 0 for first option
+ * @param[in]   onum        number of option
+ * @param[in]   value       value to encode
  *
  * @returns     amount of bytes written to @p buf
  */
-size_t coap_opt_put_block_object(uint8_t *buf, uint16_t lastonum,
-                                 coap_block1_t *block, uint16_t option);
+size_t coap_opt_put_uint(uint8_t *buf, uint16_t lastonum, uint16_t onum,
+                         uint32_t value);
```

Comment:
Shouldn't this also pass the remaining space available in `buf` so we don't silently overflow?